### DISCUSSION TITLE
Error: Cannot load library /usr/lib/tiled/plugins/libyy.so

### DISCUSSION THREAD

**Discussion by DrSensor** [mapeditor/tiled#3018] [Q&A] [answered]
Hi, somehow when I list `--export-formats` I got an error message:
```
$ tiled --export-formats
Error: Cannot load library /usr/lib/tiled/plugins/libyy.so: (/usr/lib/tiled/plugins/libyy.so: undefined symbol: _ZN2Yy10JsonWriter19writeUnquotedMemberEPKcRK10QByteArray)
Map export formats:
 csv
 gmx
 js
 json
 lua
 tmx
Tileset export formats:
 json
 lua
 tsx
```
Tiled still working fine but I'm curious what is `libyy.so`?

---
os: Archlinux
version: Tiled v1.5

**Comment by bjorn** [comment, marked as answer]
Hmm, `libyy.so` is the new GameMaker Studio 2 export plugin. The undefined symbol problem is a little unexpected, especially considering it doesn't happen in the distributed AppImage. When we demangle this symbol, we get:
```
Yy::JsonWriter::writeUnquotedMember(char const*, QByteArray const&)
```
Which is defined here:

https://github.com/mapeditor/tiled/blob/d81fe4794542d78dfaa503936e329e3520eeb18c/src/plugins/yy/jsonwriter.cpp#L178-L185

But, then I realized that Arch Linux builds are based on qmake and I checked the qmake project of the plugin:

https://github.com/mapeditor/tiled/blob/d81fe4794542d78dfaa503936e329e3520eeb18c/src/plugins/yy/yy.pro#L5

And it doesn't list `jsonwriter.cpp`. This being a library that unfortunately doesn't lead to a compile-time error, so we're running into it at runtime instead. :-(

Fortunately the fix is easy. Thanks for the report! :-)

**Reply by bjorn** [reply]
I guess a bug report would have been more appropriate for this, but unfortunately it doesn't appear GitHub provides a way to convert a discussion to an issue. I've pushed a fix in e4ba212637d4e0bfc61b98310a5c2234a92b904a, so I've marked this discussion as answered. :-)
